src/Adapters/AzureManagementAdapter.ts

```diff
@@ -567,4 +571,94 @@ export default class AzureManagementAdapter implements IAzureManagementAdapter {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return await adapterMethodSandbox.safelyFetchData(async (token) => {
+            let adtInstance: IAzureResource;
+            if (adtInstanceIdentifier.hostName) {
+                const digitalTwinInstances = await this.getResourcesByPermissions(
+                    {
+                        getResourcesParams: {
+                            resourceType: AzureResourceTypes.DigitalTwinInstance
+                        },
+                        requiredAccessRoles: RequiredAccessRoleGroupForADTInstance
+                    }
+                );
+                const result = digitalTwinInstances.result.data;
+                adtInstance = result.find(
+                    (d) =>
+                        d.properties.hostName === adtInstanceIdentifier.hostName
+                );
+            } else if (adtInstanceIdentifier.id) {
+                const digitalTwinInstance = await this.getResourceById(
+                    adtInstanceIdentifier.id
+                );
+                adtInstance = digitalTwinInstance.result.data;
+            }
+
+            try {
+                const url = `https://management.azure.com${adtInstance.id}/timeSeriesDatabaseConnections`;
+                const timeSeriesDatabaseConnections: Array<IAzureTimeSeriesDatabaseConnection> = await this.fetchAllResources(
+                    adapterMethodSandbox,
+                    [],
+                    token,
+                    url,
+                    {
+                        apiVersion:
+                            AzureResourcesAPIVersions[
+                                'Microsoft.DigitalTwins/digitalTwinsInstances/timeSeriesDatabaseConnections'
+                            ]
+                    }
+                );
+
+                const connectionData = timeSeriesDatabaseConnections[0]; // TODO: for now get the first connection information
+                const clusterUrl = connectionData.properties.adxEndpointUri;
+                const databaseName = connectionData.properties.adxDatabaseName;
+                const tableName =
+                    connectionData.properties.adxTableName ||
+                    `adt_dh_${databaseName.replace('-', '_')}_${
+                        adtInstance.location
+                    }`; // there is a default syntax used to construct the table name if not provided
+                return new ADTInstanceTimeSeriesConnectionData({
+                    kustoClusterUrl: clusterUrl,
+                    kustoDatabaseName: databaseName,
+                    kustoTableName: tableName
+                });
+            } catch (error) {
+                adapterMethodSandbox.pushError({
+                    isCatastrophic: false,
+                    rawError: error
+                });
+            }
+        }, 'azureManagement');
+    }
+
+    async getResourceById(resourceId: string) {
```

Comment:
don't we already have this API? Do we need antoher copy? Or can we not access other adapters from here?

Reply:
we didn't have `getResourceById` method in the AzureManagementAdapter before? We have been using getResources type of methods before. There is no other copy?